server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceInferOwnerTest.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Base64;
+import java.util.UUID;
+import keywhiz.IntegrationTestRule;
+import keywhiz.KeywhizService;
+import keywhiz.TestClients;
+import keywhiz.api.automation.v2.CreateGroupRequestV2;
+import keywhiz.api.automation.v2.CreateSecretRequestV2;
+import keywhiz.api.automation.v2.SecretDetailResponseV2;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SecretResourceInferOwnerTest {
+  private static final ObjectMapper mapper = KeywhizService.customizeObjectMapper(Jackson.newObjectMapper());
+  private static final Base64.Encoder encoder = Base64.getEncoder();
+  private static final Integer RESPONSE_CODE_CREATED = 201;
+
+  private static final String AUTOMATION_CLIENT_GROUP_NAME = "Security";
+  private static final String SECRET_CONTENT = "content";
+  private static final String SECRET_NAME_PREFIX = "inferOwnerTest-";
+  private static final String SECRET_OWNER = "secret owner";
+
+  private SecretResourceTest secretResourceTest;
+
+  OkHttpClient mutualSslClient;
+
+  @ClassRule public static final RuleChain chain = IntegrationTestRule.rule("keywhiz-test-newSecretOwnershipStrategyInfer.yaml");
+
+  @Before public void setUp() {
+    mutualSslClient = TestClients.mutualSslClient();
+    secretResourceTest = new SecretResourceTest();
+    secretResourceTest.mutualSslClient = mutualSslClient;
+  }
+
+  @Test
+  public void testCreateSecretWithNullOwnerName() throws IOException {
+    String secretName = SECRET_NAME_PREFIX + randomString();
+    CreateSecretRequestV2 request = CreateSecretRequestV2.builder()
+        .name(secretName)
+        .content(encode(SECRET_CONTENT))
+        .build();
+
+    Response response = secretResourceTest.create(request);
+    assertThat(response.code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    SecretDetailResponseV2 details = secretResourceTest.lookup(secretName);
+    assertThat(details.name()).isEqualTo(secretName);
+    assertThat(details.owner()).isEqualTo(AUTOMATION_CLIENT_GROUP_NAME);
+  }
+
+  @Test
+  public void testCreateSecretWithOwnerName() throws IOException {
+    String secretName = SECRET_NAME_PREFIX + randomString();
+
+    assertThat(createGroup(SECRET_OWNER).code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    CreateSecretRequestV2 request = CreateSecretRequestV2.builder()
+        .name(secretName)
+        .owner(SECRET_OWNER)
+        .content(encode(SECRET_CONTENT))
+        .build();
+
+    Response response = secretResourceTest.create(request);
+    assertThat(response.code()).isEqualTo(RESPONSE_CODE_CREATED);
+
+    SecretDetailResponseV2 details = secretResourceTest.lookup(secretName);
+    assertThat(details.name()).isEqualTo(secretName);
+    assertThat(details.owner()).isEqualTo(SECRET_OWNER);
+  }
```

Comment:
Even if you're still not sure what functionality you'll have when `clientGroups.size() != 1` (in `SecretResource`), you could still write tests for them, even if they currently fail!

Reply:
added!